infrastructure/src/main/java/org/corfudb/infrastructure/orchestrator/Orchestrator.java

```diff
@@ -75,45 +92,46 @@ void handleQuery(CorfuPayloadMsg<OrchestratorRequest> msg, ChannelHandlerContext
                 .payloadMsg(new OrchestratorResponse(resp)));
     }
 
-    void addNode(CorfuPayloadMsg<OrchestratorRequest> msg, ChannelHandlerContext ctx, IServerRouter r) {
-        AddNodeRequest req = (AddNodeRequest) msg.getPayload().getRequest();
-        if (findUUID(req.getEndpoint()) != null) {
-            // An add node workflow is already executing for this endpoint, return
+    /**
+     *
+     * Dispatch a workflow create request.
+     *
+     * Create and start a workflow on this orchestrator, if there is
+     * an existing workflow that is executing on the same endpoint,
+     * then just return the corresponding workflow id. Dispatch is the only
+     * place where workflows are created based on reading activeWorkflows
+     * and therefore needs to be synchronized.
```

Comment:
synchronized to prevent what? Multiple workflows from launching at once?

Reply:
Correct. Updated comment.